pkg/steps/pod.go

```diff
@@ -198,3 +171,82 @@ func PodStep(name string, config PodStepConfiguration, resources api.ResourceCon
 		jobSpec:     jobSpec,
 	}
 }
+
+func (s *podStep) getPodObject() (*coreapi.Pod, error) {
+	image := fmt.Sprintf("%s:%s", s.config.From.Name, s.config.From.Tag)
+	containerResources, err := resourcesFor(s.resources.RequirementsForStep(s.config.As))
+	if err != nil {
+		return nil, fmt.Errorf("unable to calculate %s pod resources for %s: %s", s.name, s.config.As, err)
+	}
+
+	if len(s.config.From.Namespace) > 0 {
+		return nil, fmt.Errorf("pod step does not supported an image stream tag reference outside the namespace")
+	}
+
+	return &coreapi.Pod{
+		ObjectMeta: meta.ObjectMeta{
+			Name: s.config.As,
+			Labels: trimLabels(map[string]string{
+				PersistsLabel:    "false",
+				JobLabel:         s.jobSpec.Job,
+				BuildIdLabel:     s.jobSpec.BuildId,
+				ProwJobIdLabel:   s.jobSpec.ProwJobID,
+				CreatedByCILabel: "true",
+			}),
+			Annotations: map[string]string{
+				JobSpecAnnotation:                     s.jobSpec.RawSpec(),
+				annotationContainersForSubTestResults: s.name,
+			},
+		},
+		Spec: coreapi.PodSpec{
+			ServiceAccountName: s.config.ServiceAccountName,
+			RestartPolicy:      coreapi.RestartPolicyNever,
+			Containers: []coreapi.Container{
+				{
+					Name:                     s.name,
+					Image:                    image,
+					Command:                  []string{"/bin/sh", "-c", "#!/bin/sh\nset -eu\n" + s.config.Commands},
+					Resources:                containerResources,
+					TerminationMessagePolicy: coreapi.TerminationMessageFallbackToLogsOnError,
+					VolumeMounts:             podVolume(s.config),
+				},
+			},
+			Volumes: podSecret(s.config),
+		},
+	}, nil
+}
+
+func podSecret(config PodStepConfiguration) []coreapi.Volume {
+	if len(config.SecretName) > 0 {
+		return []coreapi.Volume{
+			{
+				Name: testSecretName,
+				VolumeSource: coreapi.VolumeSource{
+					Secret: &coreapi.SecretVolumeSource{
+						SecretName: config.SecretName,
+					},
+				},
+			},
+		}
+	}
+	return []coreapi.Volume{}
+}
+
+func podVolume(config PodStepConfiguration) []coreapi.VolumeMount {
+	var secretPath string
+	if len(config.SecretName) > 0 {
+		if len(config.SecretPath) > 0 {
+			secretPath = config.SecretPath
+		} else {
+			secretPath = testSecretDefaultPath
+		}
+		return []coreapi.VolumeMount{
```

Comment:
since we are mounting a single secret, I think we can use also the `volumeMounts.subPath`.